test: adds automation test for checking expected replicas

jjaferson opened the pull request:
https://issues.redhat.com/browse/INTLY-5412

philbrookes commented:
/lgtm
@wei-lee any thoughts on this?

philbrookes commented:
/approve

jjaferson commented:
/retest

jjaferson commented:
/test images

b1zzu commented on test/common/deployment_types.go:
```diff
@@ -0,0 +1,261 @@
+	}
+}
+
+func getDeployment() []Namespace {
```
Why not using a global variable instant of a method?

b1zzu commented on test/common/deployment_types.go:
```diff
@@ -0,0 +1,261 @@
+					Namespace: namespace.Name,
+				},
+			}
+			err := ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: product.Name, Namespace: namespace.Name}, dc)
```
Sorry for golang ignorance but when I see a TODO for me it means that there is something that must be addressed, can I ask why?

b1zzu commented on test/common/deployment_types.go:
```diff
@@ -0,0 +1,261 @@
+			}
+			err := ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: product.Name, Namespace: namespace.Name}, dc)
+			if err != nil {
+				t.Fatalf("DeploymentConfig %s not found - %s", product.Name, err)
```
Instant of failing on the first error it would be better to check all deployment config so that if there are more than one missing we can list all of them in one single run

b1zzu commented on test/common/deployment_types.go:
```diff
@@ -0,0 +1,261 @@
+			}
+
+			if dc.Status.Replicas != product.ExpectedReplicas {
+				t.Fatalf("DeploymentConfig %s in namespace %s doens't match the number of expected replicas %s/%s",
```
Same here, see the previous comment

jjaferson commented:
/test e2e

b1zzu commented on test/common/deployment_types.go:
```diff
@@ -0,0 +1,261 @@
+	}
+}
+
+func TestStatefulSetsExpectedReplicas(t *testing.T, ctx *TestingContext) {
```
This test is not part of any AFTER_INSTALL_LIST or ALL_TEST list, therefore, is not been executed

b1zzu commented on test/common/tests.go:
```diff
@@ -6,4 +6,9 @@ var (
 		{"Verify CRD Exists", TestIntegreatlyCRDExists},
 	}
+
+	AFTER_INSTALL_TESTS = []TestCase{
```
Can I ask why a new list of tests?

b1zzu commented on test/e2e/integreatly_test.go:
```diff
@@ -100,6 +101,12 @@ func TestIntegreatly(t *testing.T) {
 			IntegreatlyCluster(t, f, ctx)
 		})
+
+		for _, test := range common.AFTER_INSTALL_TESTS {
```
If we have to create a new test list this must be also part of the functional suite

jjaferson commented:
> Hi, I have a couple of comments and ideas on how we could improve the tests, let me know wdyt

Hi davide, Thanks for reviewing. 

PR got merged but I'll submit another one addressing your points and ping you on it.
